pkg/ddevapp/silverstripe.go

```diff
@@ -0,0 +1,74 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/ddev/ddev/pkg/fileutil"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"os"
+	"path/filepath"
+)
+
+// see https://github.com/ddev/ddev/pull/5058 if the bin folder changes.
+// isSilverstripeApp returns true if the app is of type Silverstripe
+func isSilverstripeApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, app.ComposerRoot, "vendor/bin/sake"))
+}
+
+func silverstripePostStartAction(app *DdevApp) error {
+	// We won't touch env if disable_settings_management: true
+	if app.DisableSettingsManagement {
+		return nil
+	}
+	envFilePath := filepath.Join(app.AppRoot, app.ComposerRoot, ".env")
+	_, envText, err := ReadProjectEnvFile(envFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Unable to read .env file: %v", err)
+	}
+	if os.IsNotExist(err) {
+		err = fileutil.CopyFile(filepath.Join(app.AppRoot, app.ComposerRoot, ".env.example"), envFilePath)
+		if err != nil {
+			util.Debug("Silverstripe: .env.example does not exist yet, not trying to process it")
+			return nil
+		}
+		_, envText, err = ReadProjectEnvFile(envFilePath)
+		if err != nil {
+			return err
+		}
+	}
+	port := "3306"
+	dbConnection := "MySQLDatabase"
+	// Although possible, it is extremely uncommon to use Postgres with Silverstripe.
+	// Thus the option is there to override
+	if app.Database.Type == nodeps.Postgres {
+		dbConnection = "PostgreSQLDatabase"
+		port = "5432"
+	}
+	envMap := map[string]string{
+		"SS_DATABASE_SERVER":        "db",
+		"SS_DATABASE_PORT":          port,
+		"SS_DATABASE_NAME":          "db",
+		"SS_DATABASE_USERNAME":      "db",
+		"SS_DATABASE_PASSWORD":      "db",
+		"SS_ENVIRONMENT_TYPE":       "dev",
+		"SS_DATABASE_CLASS":         dbConnection,
+		"MAILER_DSN":                "smtp://localhost:1025",
```

Comment:
where does mails_dsn come from? noticed it one or two times, same as ss_database_host before, but never questioned it. but i've went to the docs page listing the available environmental variables https://docs.silverstripe.org/en/4/getting_started/environment_management/#core-environment-variables and mails_dsn isnt listed there? would it make sense to remove it as well?

Reply:
It's for Silverstripe 5:
https://docs.silverstripe.org/en/5/developer_guides/email

Reply:
ahhhh! thanks for the clarification. then it totally makes sense. 

p.s. github definitely needs a lightbulb emoji!